source/_docs/tools/dev-tools.markdown

```diff
@@ -24,3 +24,79 @@ Screenshot of Home Assistant's Developer Tools.
+
+Note that this is the state representation of a device within Home Assistant. That means, it is what Home Assistant sees, and it does not communicate with the actual device in any manner. The updated information can still be used to trigger events, and state changes. To communicate with the actual device, it is recommended to call services in the services section above, instead of updating state.
+
+For ex: Changing the `light.bedroom` state from `off` to `on` does not turn on the light. If there is an automation that triggers on the `state` change of the `light.bedroom`, it will be triggered – even though the actual bulb has not turned on. Also, when the bulb state changes – the state information will be overridden. In other words, the changes that are made through the “States” section are temporary, and is recommended to use for testing purposes only.
+
+{% linkable_title Events %}
+
+This Events section is as basic as it can get. It does only one thing – fires events on the event bus.
+To fire an event, simply type the name of the event, and pass the event data in JSON format.
+For ex: To fire a custom event, enter the `event_type` as `event_light_state_changed` and the event data JSON as 
+
+```json
+{ "state":"on" }
+```
+
+If there is an automation that handles that event, it will be automatically triggered. See below:
+```yaml
+- alias: Capture Event
+  trigger:
+    platform: event
+    event_type: event_light_state_changed
+  action:
+    - service: notify.notify
+      data_template:
+        message: "Light is turned {{ trigger.event.data.state }}"
+```
+
+{% linkable_title Template Editor %}
+
+The Template Editor provides a way to test the template code quickly. When the Template Editor page is loaded, it comes with a sample template code that illustrates how the code can be written and tested. 
+
+It has two sections, code goes on the left hand side, and the output is shown on the right hand side. The code can be removed and replaced, and when the page is loaded/refreshed, the default sample code will be loaded back. 
+
+It is a good practice to test the template code in the template editor prior to putting it in automations and scripts.
+
+For more information about jinja2, visit [jinja2 documentation](http://jinja.pocoo.org/docs/dev/templates/), and also read templating document [here](https://home-assistant.io/topics/templating/)
+
+
+{% linkable_title mqtt %}
+
+This section is only visible if the MQTT is configured. To configure MQTT, add `mqtt:` to the `configuration.yaml` file. For more information, refer to [mqtt]( https://home-assistant.io/components/mqtt/)
+Even though MQTT in general provides deeper functionality, the developer tools section of MQTT is limited to publishing messages to a given topic. It supports templates for the payload. To publish a message, simply specify the topic name and the payload and click “PUBLISH” button. 
```

Comment:
Simplify...

`Although MQTT provides greater functionality...`

Reply:
I am good with this proposal. There are many ways of saying, it, whatever makes sense to everyone!